frame/assets/src/functions.rs

```diff
@@ -179,12 +179,8 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				}
 			}
 
-			let is_provider = false;
```

Comment:
`let is_provider = details.is_sufficient;` should be the correct solution.

You are not allowed to let the account die when there is a consumer and you need to remove the provider.

Reply:
IIRC here keep_alive doesn't mean the account will die, only that the asset will go to zero. An asset is never a provider only a sufficient or consumer. So asset can always die without problem.

Reply:
Yeah good point! Maybe the code was just copied from balances.